### PR TITLE
Return 400 but not 500 when request archive with wrong format (#17691)

### PR DESCRIPTION
**Pull request by lunny**
backport #17691

### CODE REVIEW

File: integrations/api_repo_archive_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	link, _ = url.Parse(fmt.Sprintf("/api/v1/repos/%s/%s/archive/master.tar.gz", user2.Name, repo.Name))
+	link.RawQuery = url.Values{"token": {token}}.Encode()
+	resp = MakeRequest(t, NewRequest(t, "GET", link.String()), http.StatusOK)
+	bs, err = io.ReadAll(resp.Body)
+	assert.NoError(t, err)
+	assert.EqualValues(t, 266, len(bs))
+
+	link, _ = url.Parse(fmt.Sprintf("/api/v1/repos/%s/%s/archive/master.bundle", user2.Name, repo.Name))
+	link.RawQuery = url.Values{"token": {token}}.Encode()
+	resp = MakeRequest(t, NewRequest(t, "GET", link.String()), http.StatusOK)
```
Comment (Gusted) [inline integrations/api_repo_archive_test.go]:
```
api_repo_archive_test.go:43
        	Error:
        	expected: 200
        	actual  : 400
          Test    : TestAPIDownloadArchive
        	Messages:   	Request: GET /api/v1/repos/user2/repo1/archive/master.bundle?token=cdb88c06eb70795fbfaa748d1a926b562c95b825
```

File: integrations/api_repo_archive_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	bs, err = io.ReadAll(resp.Body)
+	assert.NoError(t, err)
+	assert.EqualValues(t, 382, len(bs))
```
Comment (Gusted) [inline integrations/api_repo_archive_test.go]:
```
expected: 382
actual  : 30
```

**Review by Gusted — COMMENTED** [review]
Tests seems to be failing - otherwise L G T M

**Review by KN4CK3R — APPROVED** [review]

**Comment by lunny** [comment]
@Gusted Tests PASS, so I assume this is your LGTM . :)

**Comment by lunny** [comment]
> @Gusted Tests PASS, so I assume this is your LGTM . :)

Ah, L-G-T-M bot is too smart to prevent I vote myself.

**Review by Gusted — APPROVED** [review]

**Pull request merged**